packages/devtools_app/lib/src/app.dart

```diff
@@ -277,8 +278,8 @@ class DevToolsAppState extends State<DevToolsApp> with AutoDisposeMixin {
         return DevToolsScaffold.withChild(
           key: UniqueKey(),
           ideTheme: ideTheme,
-          child: _providedControllers(
-            offline: true,
+          child: MultiProvider(
```

Comment:
@kenzieschmoll not sure if this is clearer.
Technically this doesn't solve the lint as the lint still gets mad about methods returning List<Widget> but I think it solves the spirit and in some ways is a little clearer than the opaque helper.

Reply:
The only downside of pulling the MultiProvider out of the helper is that this creates some duplicated code where we have to manually specify MultiProvider wherever _providedControllers is used, but I do see how _providedControllers already sounds like it should return a List as is. 

Why did `child: _providedControllers` trigger a lint to begin with?

Reply:
The lint is one to guide you towards creating widgets rather than creating functions that return widgets. It catches a lot of ugly code (particularly cases where the BuildContext was passed around) as well as some false positives like this.

Reply:
I wonder if the duplicate code is a little more idiomatic in that it is clear what we are doing is injecting providers.

Reply:
@incendial this was the case where I couldn't keep the code idiomatic and avoid the lint about functions returning widgets. Seems odd the lint ever fires for functions returning Lists of widgets. Notice that the lint still fires with my fix here as it is upset about the method called that returns a List of all providers for the class.

Reply:
I see, never seen this case before, tbh 😅. If I change the rule to ignore Providers, will this be enough? The more I look at this case, the more I don't understand how a possible "right way" without a method could look like.

> Seems odd the lint ever fires for functions returning Lists of widgets

Very common case is when people create a private method returning List<Widget> and then spread it in the widget's Column or Row.

Reply:
So far the only case that is really blocking me is the provider case. The Column and Row cases are interesting. I do buy that those cases are probably usually bugs.

Reply:
Thank you! Added to backlog, will ship in the next release